libs/akita/src/lib/entityStore.ts

```diff
@@ -660,6 +692,37 @@ export class EntityStore<S extends EntityState = any, EntityType = getEntityType
   private hasUIStore() {
     return this.ui instanceof EntityUIStore;
   }
+
+  protected cancelEntityTTL(ids: IDType[]) {
+    const ttl = this.getCacheTTL();
+
+    if (ttl) {
+      ids.forEach((id) => this.ttlQueue.cancel(TTLType.Entity, id as any));
+    }
+  }
+
+  protected updateEntityTTL(ids: IDType[]) {
+    const ttl = this.getCacheTTL();
+
+    if (ttl) {
+      ids.forEach((id) => this.ttlQueue.update(ttl, TTLType.Entity, id as any));
+    }
+  }
+
+  protected watchCacheTTL() {
+    super.watchCacheTTL();
+    this.ttlQueue.expired$.pipe(filter((ttl) => ttl.type === TTLType.Entity)).subscribe((ttl) => {
+      if (ttl.id && this.ids.includes(ttl.id)) {
+        this._setState((state) => {
+          const idsExpired = state.idsExpired || [];
+          return {
+            ...state,
+            idsExpired: idsExpired.includes(ttl.id) ? idsExpired : [...idsExpired, ttl.id],
```

Comment:
This will trigger a redundant dispatch in case of `idsExpired.includes(ttl.id)`.